api/tests/DataPersister/CampDataPersisterTest.php

```diff
@@ -88,4 +92,40 @@ public function testDoesNotSetCreatorAndOwnerOnUpdate() {
         $this->assertNotEquals($user, $data->creator);
         $this->assertNotEquals($user, $data->owner);
     }
+
+    public function testCreatesCampCollaborationOnCreate() {
+        // given
+        $user = new User();
+        $this->security->method('getUser')->willReturn($user);
+        $this->decoratedMock->expects($this->once())->method('persist')->willReturnArgument(0);
+
+        // then
+        $this->em->expects($this->once())->method('persist')->will($this->returnCallback(function ($object) use ($user) {
+            $this->assertInstanceOf(CampCollaboration::class, $object);
+            /** @var CampCollaboration $campCollaboration */
+            $campCollaboration = $object;
+            $this->assertEquals($user, $campCollaboration->user);
+            $this->assertEquals($this->camp, $campCollaboration->camp);
+            $this->assertEquals(CampCollaboration::STATUS_ESTABLISHED, $campCollaboration->status);
+            $this->assertEquals(CampCollaboration::ROLE_MANAGER, $campCollaboration->role);
+        }));
+
+        // when
+        /** @var Camp $data */
+        $data = $this->dataPersister->persist($this->camp, ['collection_operation_name' => 'post']);
+    }
+
+    public function testDoesNotCreateCampCollaborationOnCreate() {
```

Comment:
should this be called `testDoesNotCreateCampCollaborationOnUpdate`?

Reply:
Nice catch! Merci